File: Python/Exercicios_diarios/Revisão_intermediaria/Cordas isomórficas.cpp
#include <iostream>
using namespace std;

class Solution
{
public:
    bool isIsomorphic(string str_1, string str_2)
    {
        int aux_1[256] = {0}, aux_2[256] = {0};

        for (int i = 0; i < str_1.size(); ++i)
        {
            if (aux_1[str_1[i]] != aux_2[str_2[i]])
                return false;

            aux_1[str_1[i]] = i + 1;
            aux_2[str_2[i]] = i + 1;
        }
        
        return true;
    }
};

File: Python/Exercicios_diarios/Revisão_intermediaria/Cordas isomórficas.py
class Solution(object):
    def isIsomorphic(self, s, t):
        s_t, t_s = {}, {}
        
        for i in range(len(s)):
            if s[i] in s_t and s_t[s[i]] != t[i]:
                return False
            if t[i] in t_s and t_s[t[i]] != s[i]:
                return False
            
            s_t[s[i]] = t[i]
            t_s[t[i]] = s[i]
            
        return True
